testinfra/modules/iptables.py

```diff
@@ -0,0 +1,87 @@
+# coding: utf-8
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import unicode_literals
+
+from testinfra.modules.base import Module
+
+
+class Iptables(Module):
+    """Test iptables rule exists"""
+
+    def __init__(self, rule=None, table=None):
+        self.rule = rule
+        self.table = table
+        super(Iptables, self).__init__()
+
+    @property
+    def exists(self):
+        """Returns true if iptables rule exists in table
+
+        Defaults to looking in 'filter' table.
+
+        >>> host.iptables_rule("-A INPUT -j REJECT").exists
+        True
+        >>> host.iptables_rule("-A INPUT -i lo -j REJECT").exists
+        False
+        >>> host.iptables_rule(
+            "-A PREROUTING -d 192.168.0.1/32 -j REDIRECT",
+            "nat"
+        ).exists
+        True
+
+        """
+
+        rules = self.get_rules()
+        return self.rule in rules
+
+    def get_rules(self, table='filter', chain=None):
+        """Returns list of iptables rules by running
```

Comment:
Looks like the `by running` is the beginning of an unfinished sentence, I'm wrong ?

Reply:
fixed